packages/webapp/src/containers/Log/HarvestUseType/index.js

```diff
@@ -26,40 +26,57 @@ function HarvestUseType() {
   useEffect(() => {}, []);
 
   const onBack = (data) => {
+    dispatch(canEditStepTwo(false));
     const tempProps = JSON.parse(JSON.stringify(data));
+
     if (defaultData.selectedUseTypes.length > 0) {
-      tempProps.selectedUseTypes.map((item, idx) => {
-        if (
-          idx < defaultData.selectedUseTypes.length &&
-          item.harvest_use_type_name === defaultData.selectedUseTypes[idx].harvest_use_type_name
-        ) {
-          item.quantity_kg = defaultData.selectedUseTypes[idx].quantity_kg;
-        }
+      tempProps.selectedUseTypes.map((item) => {
+        defaultData.selectedUseTypes.map((item1) => {
+          let name = isEditStepTwo.isEditStepTwo
+            ? item1.harvestUseType.harvest_use_type_name
+            : item1.harvest_use_type_name;
+          if (item.harvest_use_type_name === name) {
+            item.quantity_kg = item1.quantity_kg;
+          }
+        });
       });
     }
     defaultData.selectedUseTypes = tempProps.selectedUseTypes;
-    dispatch(canEditStepTwo(false));
     dispatch(harvestLogData(defaultData));
     history.push('/harvest_log');
   };
 
   const onNext = (data) => {
-    const tempProps = JSON.parse(JSON.stringify(data));
-    if (defaultData.selectedUseTypes.length > 0) {
+    if (isEditStepTwo.isEditStepTwo) {
+      data.selectedUseTypes.map((item) => {
+        defaultData.selectedUseTypes.map((item1) => {
+          if (item.harvest_use_type_name === item1.harvestUseType.harvest_use_type_name) {
+            item.quantity_kg = item1.quantity_kg;
+          }
+        });
```

Comment:
Try to do the same as above. In fact if you could wrap the reduce function and the map function into separate functions that would be great.